From 79758df3803e6d199a277820333e7078fd01d8a0 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 18 Mar 2023 21:35:07 +0000
Subject: [PATCH 1/3] gnu: Add cl-confidence.

* gnu/packages/lisp-check.scm (cl-confidence, ecl-confidence,
  sbcl-confidence): New variables.
---
 gnu/packages/lisp-check.scm | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/gnu/packages/lisp-check.scm b/gnu/packages/lisp-check.scm
index 172f5b3873..ed9c4ef568 100644
--- a/gnu/packages/lisp-check.scm
+++ b/gnu/packages/lisp-check.scm
@@ -378,6 +378,40 @@ (define-public cl-clunit2
 (define-public ecl-clunit2
   (sbcl-package->ecl-package sbcl-clunit2))
 
+(define-public sbcl-confidence
+  (let ((commit "5cbc74715348e12e689afb2d459dcb216c640a44")
+        (revision "0"))
+    (package
+      (name "sbcl-confidence")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/melusina-org/cl-confidence")
+               (commit commit)))
+         (file-name (git-file-name "cl-confidence" version))
+         (sha256
+          (base32 "0zc135rvq2valrw15bh8k6i53v7kk5l7x0kccb1bf7pglc8zgivs"))))
+      (build-system asdf-build-system/sbcl)
+      (arguments
+       `(#:asd-systems '("org.melusina.confidence"
+                         "org.melusina.confidence/testsuite")))
+      (inputs (list sbcl-alexandria))
+      (home-page "https://github.com/melusina-org/cl-confidence")
+      (synopsis "Interactive Test Framework for Common Lisp")
+      (description
+       "Confidence is a test framework for Common Lisp that focuses on simplicty. It
+avoids bureacracy and makes it easy to work interactively, without a complicated
+setup, and with explicit functions and decisions.")
+      (license license:expat))))
+
+(define-public ecl-confidence
+  (sbcl-package->ecl-package sbcl-confidence))
+
+(define-public cl-confidence
+  (sbcl-package->cl-source-package sbcl-confidence))
+
 (define-public sbcl-eos
   (let ((commit "b4413bccc4d142cbe1bf49516c3a0a22c9d99243")
         (revision "2"))
-- 
2.39.2

